packages/headless/src/features/generated-answer/generated-answer-slice.ts

```diff
@@ -68,12 +75,19 @@ export const generatedAnswerReducer = createReducer(
         state.feedbackSubmitted = true;
       })
       .addCase(resetAnswer, (state) => {
-        return {
-          ...getGeneratedAnswerInitialState(),
-          responseFormat: state.responseFormat,
-          fieldsToIncludeInCitations: state.fieldsToIncludeInCitations,
-          isVisible: state.isVisible,
-        };
+        return handleResetAnswer(state);
+      })
+      .addCase(executeSearch.fulfilled, (state, payload) => {
+        const streamId =
+          payload.payload.response.extendedResults
+            .generativeQuestionAnsweringId;
+        return state.currentStreamId === streamId
```

Comment:
This is the important line:
We moved the 'current-to-next' comparison to the reducer, which allows us to remove storage of state in in the ASM.